Week2/js/HomeTask2.js

```diff
@@ -0,0 +1,269 @@
+    address:
+      "TORONTO 5 Park Home Avenue, Suite 400, ON M2N 6L4, North York, Toronto Canada"
+  },
+  {
+    name: "Hungary",
+    logopath: "./Assets/hungary.png",
+    address: `FRANKFURT AM MAIN
+    Franklinstrasse 56,
+    60486 Frankfurt am Main,
+    Germany`
+  },
+  {
+    name: "Ireland",
+    logopath: "./Assets/ireland.png",
+    address: `DUBLIN
+    Alexandra House,
+    The Sweepstakes,
+    Ballsbridge, Dublin 4,
+    D04 C7H2
+    Ireland`
+  },
+  {
+    name: "Netherlands",
+    logopath: "./Assets/netherlands.png",
+    address: `DELFT
+    Delftechpark 37j
+    2628 XJ Delft
+    Netherlands`
+  }
+];
+function createPage() {
+  // navbar
+  let header = document.createElement("div");
+  header.setAttribute("class", "header");
+  let title = document.createElement("h1");
+  title.innerText = "NEWSFEED";
+  title.setAttribute("class", "newsfeed");
+  let another_feedback = document.createElement("h4");
+  another_feedback.className = "another-feedback";
+  another_feedback.innerText = "Yet another feedback";
+  header.appendChild(title);
+  header.appendChild(another_feedback);
+  document.body.appendChild(header);
+  // navbar end
+  createContent();
+}
+
+// this method creates the main content of the body
+var createContent = function() {
+  mainDivision.setAttribute("class", "maindiv");
+  mainDivision.setAttribute("id", "maindiv");
+
+  let mainContainer = document.createElement("div");
+  mainContainer.setAttribute("class", "flex-container");
+  let logo = document.createElement("div");
+  logo.setAttribute("class", "logo");
+  let content = document.createElement("div");
+  content.setAttribute("class", "content");
+  mainContainer.appendChild(logo);
+  mainContainer.appendChild(content);
+
+  let category = document.createElement("div");
+  category.setAttribute("class", "category");
+  let categories = createCategories();
+  let emailInputField = document.createElement("input");
+  emailInputField.setAttribute("type", "text");
+  emailInputField.setAttribute("name", "fname");
+  emailInputField.setAttribute("class", "input");
+  emailInputField.setAttribute("id", "email");
+  emailInputField.addEventListener("blur", validateEmailId);
+  let subscribe = document.createElement("button");
+  subscribe.setAttribute("value", "Subscribe");
+  subscribe.setAttribute("class", "subscribe-button");
+  subscribe.innerText = "Subscribe";
+  let errorMessage = document.createElement("p");
+  errorMessage.setAttribute("id", "error");
+  category.appendChild(categories);
+  category.appendChild(emailInputField);
+  category.appendChild(subscribe);
+  category.appendChild(errorMessage);
+  mainContainer.appendChild(category);
+  mainDivision.appendChild(mainContainer);
+  dataDiv.setAttribute("id", "data-div");
+  displayAll(dataDiv);
+  document.body.appendChild(mainDivision);
+};
+
+var createPopUp = function() {
+  let popUpContainer = document.createElement("div");
+  popUpContainer.setAttribute("id", "popup");

```

Comment:
You are creating a popup for for each click event listener of continue reading button. Instead you should create a popup initially and hide it. Once we click on the button you should pass the text to it and show. On click on close you should close that.

Currently when you click on continue reading button on different items, to close the popup we have to click n number of times.